.github/workflows/new-contributor.yml

```diff
@@ -0,0 +1,45 @@
+name: New Contributor Action
+on: 
+    pull_request_target:
+        types: [closed]
+
+concurrency:
+    group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+    cancel-in-progress: true
+
+jobs:
+  check-new-contributor:
+    if: ${{ github.event.pull_request.merged }}
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+      - name: Check new contributor
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea
+        with:
+          script: |
+            const prAuthor = context.payload.pull_request.user.login;
+            const prMerger = context.payload.pull_request.merged_by.login;
+            const repoOwner = context.repo.owner;
+            const repoName = context.repo.repo;
+            const currentPR = context.payload.pull_request.number;
+            const mergedPRs = await github.paginate(
+              github.rest.pulls.list,
+              {
+                owner: repoOwner,
+                repo: repoName,
+                state: 'closed',
+                per_page: 100
+              },
+              (response) => response.data.filter(pr => pr.merged_at != null && pr.user.login === prAuthor && pr.number !== currentPR)
+            );
```

Comment:
Do correct me if I am wrong, from my understanding this gets the last 100 pull requests and checks if the user has a merged PR within the last 100 PRs, would this mean that the message will be sent for a user that hasn't had any merged PRs in the last 100 PRs?

Reply:
Hi @itsyme, It collects all the PRs, its just that it collects them 100 at a time for the pagination. You can refer to [here](https://docs.github.com/en/rest/using-the-rest-api/using-pagination-in-the-rest-api?apiVersion=2022-11-28#example-using-the-octokitjs-pagination-method)